mesonbuild/build.py

```diff
@@ -863,6 +863,8 @@ def get_external_deps(self):
 
     def link(self, target):
         for t in listify(target, unholder=True):
+            if not isinstance(t, Target):
+                raise InvalidArguments('{!r} is not a linkable target.'.format(t))
```

Comment:
Perhaps make the error message match the test: "is not a target" or "is not a build_target" ?  We check if it's linkable next...